version: '3.8'

services:
  mongodb:
    image: mongo:latest
    container_name: mongodb
    ports:
      - "27017:27017"
    volumes:
      - mongo-data:/data/db
    networks:
      - shared_network

  recovery-app:
    build: .
    ports:
      - "8081:8080"
    depends_on:
      - db
    env_file:
      - .env
    networks:
      - shared_network

  zookeeper:
    image: wurstmeister/zookeeper:3.7.0
    container_name: zookeeper
    ports:
      - "2181:2181"
    volumes:
      - zookeeper-data:/data
    networks:
      - shared_network

  kafka:
    image: wurstmeister/kafka:latest
    container_name: kafka
    depends_on:
      - zookeeper
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_LISTENERS: INSIDE://kafka:9093,OUTSIDE://localhost:9092
      KAFKA_LISTENER_NAMES: INSIDE,OUTSIDE
      KAFKA_LISTENERS: INSIDE://0.0.0.0:9093,OUTSIDE://0.0.0.0:9092
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
      KAFKA_LISTENER_NAME: INSIDE
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
    volumes:
      - kafka-data:/var/lib/kafka
    networks:
      - shared_network

volumes:
  mongo-data:
  zookeeper-data:
  kafka-data:

networks:
  shared_network:
    external: true